Source/Core/Core/ScriptEngine.cpp

```diff
@@ -0,0 +1,945 @@
+{
+  const u8* idx = reinterpret_cast<const u8*>(lua_touserdata(L, 1));
+  lua_Integer idx_arg = luaL_checkinteger(L, 2);
+  PowerPC::PairedSingle& ps = PowerPC::ppcState.ps[*idx];
+  u32 value;
+  if (idx_arg == 0)
+    value = ps.PS0AsU32();
+  else if (idx_arg == 1)
+    value = ps.PS1AsU32();
+  else
+    return 0;
+  lua_pushinteger(L, static_cast<lua_Integer>(value));
+  return 1;
+}
+
+// Lua: set_ps_u32(userdata ps_reg, number idx, number value) -> ()
+// Sets one of the two elements of a PS (paired single) register in u32 mode.
+static int set_ps_u32(lua_State* L)
+{
+  const u8* idx = reinterpret_cast<const u8*>(lua_touserdata(L, 1));
+  lua_Integer idx_arg = luaL_checkinteger(L, 2);
+  lua_Integer value_arg = luaL_checkinteger(L, 3);
+  u32 value_32 = static_cast<u32>(value_arg);
+  u64 value = static_cast<u64>(value_32);
+  PowerPC::PairedSingle& ps = PowerPC::ppcState.ps[*idx];
+  if (idx_arg == 0)
+    ps.SetPS0(value);
+  else if (idx_arg == 1)
+    ps.SetPS1(value);
+  return 0;
+}
+
+// Lua: get_ps_f64(userdata ps_reg, number idx) -> (number)
+// Accesses one of the two elements of a PS (paired single) register in f64 mode.
+static int get_ps_f64(lua_State* L)
+{
+  const u8* idx = reinterpret_cast<const u8*>(lua_touserdata(L, 1));
+  lua_Integer idx_arg = luaL_checkinteger(L, 2);
+  PowerPC::PairedSingle& ps = PowerPC::ppcState.ps[*idx];
+  double value;
+  if (idx_arg == 0)
+    value = ps.PS0AsDouble();
+  else if (idx_arg == 1)
+    value = ps.PS1AsDouble();
+  else
+    return 0;
+  lua_pushnumber(L, value);
+  return 1;
+}
+
+// Lua: set_ps_f64(userdata ps_reg, number idx, number value) -> ()
+// Sets one of the two elements of a PS (paired single) register in f64 mode.
+static int set_ps_f64(lua_State* L)
+{
+  const u8* idx = reinterpret_cast<const u8*>(lua_touserdata(L, 1));
+  lua_Integer idx_arg = luaL_checkinteger(L, 2);
+  lua_Number value = luaL_checkinteger(L, 3);
+  PowerPC::PairedSingle& ps = PowerPC::ppcState.ps[*idx];
+  if (idx_arg == 0)
+    ps.SetPS0(value);
+  else if (idx_arg == 1)
+    ps.SetPS1(value);
+  return 0;
+}
+
+// Converts a string with max 8 chars to u64.
+static constexpr u64 string_to_u64(const std::string_view& e)
+{
+  u64 res = 0;
+  if (e.length() >= 1)
+    res |= (static_cast<u64>(e[0]) << 56u);
+  if (e.length() >= 2)
+    res |= (static_cast<u64>(e[1]) << 48u);
+  if (e.length() >= 3)
+    res |= (static_cast<u64>(e[2]) << 40u);
+  if (e.length() >= 4)
+    res |= (static_cast<u64>(e[3]) << 32u);
+  if (e.length() >= 5)
+    res |= (static_cast<u64>(e[4]) << 24u);
+  if (e.length() >= 6)
+    res |= (static_cast<u64>(e[5]) << 16u);
+  if (e.length() >= 7)
+    res |= (static_cast<u64>(e[6]) << 8u);
+  if (e.length() >= 8)
+    res |= static_cast<u64>(e[7]);
+  if (e.length() >= 9)
+    res = 0;
+  return res;
+}
+
+// Lua: ppc_get(table ppc, string field) -> (any)
+// Gets a field or register on the special "ppc" table.
+static inline int ppc_get(lua_State* L)
```

Comment:
```suggestion
static int ppc_get(lua_State* L)
```

inline does nothing here, given the internal linkage